runtime/src/epoch_stakes.rs

```diff
@@ -152,11 +153,76 @@ impl From<VersionedEpochStakes> for EpochStakes {
     }
 }
 
+/// Only the `StakesEnum::Delegations` variant is unable to be serialized as a
+/// `StakesEnum::Stakes` variant, so leave those entries and split off the other
+/// epoch stakes enum variants into a new map which will be serialized into the
+/// new `versioned_epoch_stakes` snapshot field.  After a cluster transitions to
+/// serializing epoch stakes in the new format, `StakesEnum::Delegations`
+/// variants for recent epochs will no longer be created and can be deprecated.
+pub(crate) fn split_epoch_stakes(
+    bank_epoch_stakes: HashMap<Epoch, EpochStakes>,
+) -> (
+    HashMap<Epoch, EpochStakes>,
+    HashMap<Epoch, VersionedEpochStakes>,
+) {
+    let mut old_epoch_stakes = HashMap::new();
+    let mut versioned_epoch_stakes = HashMap::new();
+    for (epoch, epoch_stakes) in bank_epoch_stakes.into_iter() {
+        let EpochStakes {
+            stakes,
+            total_stake,
+            node_id_to_vote_accounts,
+            epoch_authorized_voters,
+        } = epoch_stakes;
+        match stakes.as_ref() {
+            StakesEnum::Delegations(_) => {
+                old_epoch_stakes.insert(
+                    epoch,
+                    EpochStakes {
+                        stakes: stakes.clone(),
+                        total_stake,
+                        node_id_to_vote_accounts,
+                        epoch_authorized_voters,
+                    },
+                );
+            }
+            StakesEnum::Accounts(stakes) => {
+                versioned_epoch_stakes.insert(
+                    epoch,
+                    VersionedEpochStakes::Current {
+                        stakes: Stakes::<Stake>::from(stakes.clone()),
```

Comment:
@brooksprumo as you pointed out, this is causing us to do a clone of all epoch stakes. I see a few workarounds:
1. replace all of the bank epoch stakes entries with `StakesEnum::Stakes` when we compute them for snapshots
2. when we save off bank epoch stakes during epoch boundaries, replace the `StakesEnum::Accounts` with `StakesEnum::Stakes` immediately
3. do the custom serialization approach we do for the `epoch_stakes` field where we keep `StakesEnum::Accounts` in `VersionedEpochStakes` but when serializing, it gets serialized as a `StakesEnum::Stakes`

I'm going to explore the first one